src/Spice86.Core/Emulator/Devices/ExternalInput/IHardwareInterruptController.cs

```diff
@@ -0,0 +1,56 @@
+namespace Spice86.Core.Emulator.Devices.ExternalInput;
+
+using Spice86.Shared.Emulator.Errors;
+
+/// <summary>
+/// Controller to route external irq to cpu interrupt vectors. 
+/// </summary>
+public interface IHardwareInterruptController {
+    /// <summary>
+    /// Services an IRQ request
+    /// </summary>
+    /// <param name="irq">The IRQ Number, which will be internally translated to a vector number</param>
+    /// <exception cref="UnrecoverableException">If not defined in the ISA bus IRQ table</exception>
```

Comment:
This exception documentation is out of date (I know that because I wrote it) :)

Reply:
Removed